### PR TITLE
Fix metrics logging error; Remove version log

### PR DESCRIPTION
**Pull request by jdn5126**
*Issue #, if available:*
https://github.com/aws/aws-network-policy-agent/issues/103
https://github.com/aws/aws-network-policy-agent/issues/49

*Description of changes:*
This PR fixes two issues:
1. The metrics server runs in a goroutine. Logging was broken in the metrics server because the call stack was starting at -2 when it needed to start at 0. This PR instantiates a new logger in the metrics goroutine and resolves https://github.com/aws/aws-network-policy-agent/issues/103
2. During init, the agent was trying to log version information. This information is not properly set, leading to https://github.com/aws/aws-network-policy-agent/issues/49. This PR removes that log, as the image tag is sufficient to determine version.

New log looks like:
```
% kubectl logs -n kube-system aws-node-57t4f -c aws-eks-nodeagent
{"level":"info","ts":"2023-12-26T20:57:29.306Z","caller":"metrics/metrics.go:23","msg":"Serving metrics on ","port":61680}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -56,11 +55,6 @@ func init() {
 
 func main() {
 	initLogger, _ := getLoggerWithLogLevel("info", "")
-	initLogger.Info("version",
-		"GitVersion", version.GitVersion,
```
Comment (jayanthvn) [inline main.go:60]: Yeah we never rebuild the same tag so just the tag should be sufficient to identify the commit..

**Review by jayanthvn — APPROVED** [review]
lgtm

File: main.go [hunk 2]
```diff
@@ -135,6 +129,6 @@ func loadControllerConfig() (config.ControllerConfig, error) {
 
 // getLoggerWithLogLevel returns logger with specific log level.
 func getLoggerWithLogLevel(logLevel string, logFilePath string) (logr.Logger, error) {
-	ctrlLogger := logger.New(logLevel, logFilePath)
+	ctrlLogger := logger.New(logLevel, logFilePath, 2)
```
Comment (jayanthvn) [inline main.go:132]: Nit: later we can move the logger defaults of 0,1,2..to utils as enum

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -43,3 +45,8 @@ func setupMetricsServer() *http.Server {
 	}
 	return server
 }
+
+func getMetricsLogger() logr.Logger {
+	ctrlLogger := logger.New("info", "", 0)
```
Comment (jayanthvn) [inline pkg/metrics/metrics.go:50]: Nit: Same as above..this can be a util enum..but can take it up as followup..

File: main.go (hunk 2 above)
Comment (jdn5126) [reply main.go:132]: Yeah, I considered an enum, but could not come up with any names that make sense, as the value to use is dependent on where the logger will be called from. If you are within a goroutine function, you want 0, as above is the standard library runtime. If you are within the controller handle routines, you want 2, as you want to skip the controller call stack surrounding the function

**Pull request merged**